server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -130,18 +204,30 @@ void doRun() {
             return;
         }
 
-        // We are only targeting in-flight cancellation of SearchShardTask for now.
-        List<SearchShardTask> searchShardTasks = getSearchShardTasks();
+        List<CancellableTask> searchTasks = getTaskByType(SearchTask.class);
+        List<CancellableTask> searchShardTasks = getTaskByType(SearchShardTask.class);
+        List<CancellableTask> cancellableTasks = new ArrayList<>();
 
         // Force-refresh usage stats of these tasks before making a cancellation decision.
+        taskResourceTrackingService.refreshResourceStats(searchTasks.toArray(new Task[0]));
         taskResourceTrackingService.refreshResourceStats(searchShardTasks.toArray(new Task[0]));
 
-        // Skip cancellation if the increase in heap usage is not due to search requests.
-        if (isHeapUsageDominatedBySearch(searchShardTasks) == false) {
+        // Check if increase in heap usage is due to SearchTasks
+        if (isHeapUsageDominatedBySearch(searchTasks, getSettings().getSearchTaskSettings().getTotalHeapBytesThreshold())) {
+            cancellableTasks.addAll(searchTasks);
+        }
+
+        // Check if increase in heap usage is due to SearchShardTasks
+        if (isHeapUsageDominatedBySearch(searchShardTasks, getSettings().getSearchShardTaskSettings().getTotalHeapBytesThreshold())) {
+            cancellableTasks.addAll(searchShardTasks);
+        }
+
+        // none of the task type is breaching the heap usage thresholds and hence we do not cancel any tasks
+        if (cancellableTasks.isEmpty()) {
             return;
         }
 
-        for (TaskCancellation taskCancellation : getTaskCancellations(searchShardTasks)) {
+        for (TaskCancellation taskCancellation : getTaskCancellations(cancellableTasks)) {
```

Comment:
Not directly related to this change, but a general question.
Seems like `getTaskCancellations` itself increments the `cancellationCount` in the tracker but only after that we check the `rateLimiter`. So in case if `rateLimiter` says limit reached, then does the cancellation count still get incremented? Could you please confirm the behavior here?